dev-client/src/screens/LocationScreens/components/soilId/SoilIdMatchesSection.tsx

```diff
@@ -65,26 +66,23 @@ export const SoilIdMatchesSection = ({
           <TopSoilMatchesInfoContent isSite={isSite} />
         </InfoButton>
       </Row>
+      <RestrictByConnectivity offline={true}>
+        <OfflineMessageBox message={t('site.soil_id.matches.offline')} />
+      </RestrictByConnectivity>
       <MatchTilesOrMessage siteId={siteId} coords={coords} />
     </ScreenContentSection>
   );
 };
 
 const MatchTilesOrMessage = ({siteId, coords}: SoilIdMatchesSectionProps) => {
   const isOffline = useIsOffline();
-  const {t} = useTranslation();
-
   const soilIdData = useSoilIdData(coords, siteId);
   const status = soilIdData.status;
   const isSite = !!siteId;
 
-  if (isOffline) {
-    return <OfflineMessageBox message={t('site.soil_id.matches.offline')} />;
-  }
-
   switch (status) {
     case 'loading':
-      return <ActivityIndicator size="small" />;
```

Comment:
Would we ever be in 'loading' state and offline? If so would we want to avoid showing the `ActivityIndicator`?

Reply:
This is what the below does -- the indicator is hidden when offline.

Reply:
Ohh yup, I'd misread it, thanks!